app/src/test/java/com/epfl/neighborfood/neighborfoodandroid/database/DatabaseFactoryTest.java

```diff
@@ -0,0 +1,50 @@
+package com.epfl.neighborfood.neighborfoodandroid.database;
+
+import static org.junit.Assert.assertEquals;
+
+import com.epfl.neighborfood.neighborfoodandroid.database.DBData;
+import com.epfl.neighborfood.neighborfoodandroid.database.DBDataCollection;
+import com.epfl.neighborfood.neighborfoodandroid.database.DatabaseFactory;
+import com.epfl.neighborfood.neighborfoodandroid.database.TentativeDB;
+import com.google.android.gms.tasks.Task;
+
+import org.junit.Test;
+
+import java.util.Map;
+
+public class DatabaseFactoryTest {
+
+    @Test
+    public void dependencyTest() {
+
+        TentativeDB dep = new TentativeDB() {
+            @Override
+            public Task<DBData> fetch(String collectionPath, String documentPath) {
+                return null;
+            }
+
+            @Override
+            public Task<Void> set(String collectionPath, String documentPath, Map<String, Object> data) {
+                return null;
+            }
+
+            @Override
+            public Task<Void> delete(String collectionPath, String documentPath) {
+                return null;
+            }
+
+            @Override
+            public Task<String> add(String collectionPath, Map<String, Object> data) {
+                return null;
+            }
+
+            @Override
+            public Task<DBDataCollection> fetchAll(String collectionPath) {
+                return null;
+            }
+        };
+
+        //DatabaseFactory.setDependency(dep);
```

Comment:
Why is there commented out code ?

Reply:
Whenever Firebase is called within a unit test, the tests crashes. I am still looking for an explanation or a workaround , but this won't be done this sprint